awriter/writer.go

```diff
@@ -66,6 +66,10 @@ func calcDataHash(manifestChecksumStore *artifact.ChecksumStore, upd *Updates, a
 	for i, u := range updates {
 		var files [](*handlers.DataFile)
 		if augmented {
+			if u == nil {
```

Comment:
Can you please elaborate a little bit? The augmented bool is set, as an augmented part is present, but it contains no data?

Reply:
The function is always called, whether there are augmented sections or not. Think of the bool more as "enable augmented mode" than "there are augmented sections". So here we check whether there is, in fact, an augmented part in this payload. It's not practical to check beforehand, because you need to traverse the whole slice to know that all of them are nil.